libkbfs/config_local.go

```diff
@@ -570,22 +570,22 @@ func (c *ConfigLocal) ResetCaches() {
 	c.kcache = NewKeyCacheStandard(5000)
 	// Limit the block cache to 10K entries or 1024 blocks (currently 512MiB)
 	c.bcache = NewBlockCacheStandard(c, 10000, MaxBlockSizeBytesDefault*1024)
-	// Limit the number of unsynced (or actively syncing) bytes to 5
-	// MB (aka, the number of parallel block puts times the max size
-	// of a block).
-	unsyncedDirtyBytesLimit := int64(5 << 20)
-	// Limit the number of total dirty bytes (including those blocks
-	// that have already finished syncing, but for which the overall
-	// Sync operation isn't yet done) to 10 MB.
-	totalDirtyBytesLimit := 2 * unsyncedDirtyBytesLimit
-	c.dirtyBcache = NewDirtyBlockCacheStandard(unsyncedDirtyBytesLimit,
-		totalDirtyBytesLimit)
+	minSyncBufferSize := int64(MaxBlockSizeBytesDefault * 10)
+	// The maximum number of bytes we can try to sync at once (also
+	// limits the amount of memory used by dirty blocks).
+	maxSyncBufferSize :=
+		int64(MaxBlockSizeBytesDefault * maxParallelBlockPuts * 2)
```

Comment:
Good point about max being smaller than min, I will fix that.

The 2x here is mostly a gut feeling I have.  If we have exactly `maxParallelBlockPuts`, then every time one block `Put` finishes, we'll effectively be wasting a slot while we wait for the other outstanding puts to remain.  If the connection can handle it without timing out, we should parallelize even more by using that slot for another block put instead -- hence making the buffer twice as big.  Maybe 1.5x would make more sense?

I can explain that in a comment, or change it, if you think something else would make more sense.